packages/devtools_app/lib/src/screens/deep_link_validation/deep_links_controller.dart

```diff
@@ -459,7 +470,17 @@ class DeepLinksController extends DisposableController {
           displayOptionsNotifier.value.updateFilter(removedFilter, false);
     }
 
-    displayLinkDatasNotifier.value = getFilterredLinks(allValidatedLinkDatas!);
+    if (addedFilter != null || removedFilter != null) {
+      applyFilters();
+    }
+  }
+
+  void applyFilters() {
+    displayLinkDatasNotifier.value = getFilterredLinks(allValidatedLinkDatas);
+    linkDataByDomainNotifier.value =
+        getFilterredLinks(allValidatedLinkDatasbyDomain);
+    linkDataByPathNotifier.value =
+        getFilterredLinks(allValidatedLinkDatasbyPath);
```

Comment:
since these three seem to always be updated in tandem, what about creating a dart object that contains all three, and then we can have a single value notifier instead of three? Or we could use a record with named elements if we don't want to create another dart class:

```
typedef ValidatedLinkData = ({List<LinkData> all, List<LinkData> byDomain, List<LinkData> byPath});
...
final validatedLinkData = (all: [], byDomain: [], byPath: []);
final filteredValidatedLinkData = ValueNotifier<ValidatedLinkData>((all: [], byDomain: [], byPath: []);
```

Reply:
done